sdk/eventhub/azure-eventhub/tests/pyamqp_tests/unittest/test_link.py

```diff
@@ -111,4 +111,45 @@ def test_receive_transfer_continuation_frame():
     assert link.delivery_count == 1
     link._incoming_transfer(transfer_frame_three)
     assert link.current_link_credit == 1
-    assert link.delivery_count == 2
\ No newline at end of file
+    assert link.delivery_count == 2
+
+
+def test_receive_transfer_and_flow():
+    def mock_outgoing(): pass
+    session = None
+    link = ReceiverLink(
+        session,
+        3,
+        source_address="test_source",
+        target_address="test_target",
+        network_trace=False,
+        network_trace_params={},
+        on_transfer=Mock(),
+    )
+
+    link._outgoing_flow =  mock_outgoing
+    link.total_link_credit = 0 # Set the total link credit to 0 to start, no credit on the wire
+
+    link.flow(link_credit=100) # Send a flow frame with desired link credit of 100
+
+    # frame: handle, delivery_id, delivery_tag, message_format, settled, more, rcv_settle_mode, state, resume, aborted, batchable, payload
+    transfer_frame_one = [3, 0, b'/blah', 0, True, False, None, None, None, None, False, ""]
+    transfer_frame_two = [3, 1, b'/blah', 0, True, False, None, None, None, None, False, ""]
+    transfer_frame_three = [3, 2, b'/blah', 0, True, False, None, None, None, None, False, ""]
+
+    link._incoming_transfer(transfer_frame_one)
+    assert link.current_link_credit == 99
+    assert link.total_link_credit == 99
+
+    # Only received 1 transfer frame per receive call, we set desired link credit again
+    # this will send a flow of 1
+    link.flow(link_credit=100)
```

Comment:
can check calling flow without passing link_credit? also, is this flow used/similar when sending?

Reply:
flow() without passing link credit would be 0 when receiving so nothing would happen. Flow() is not used when sending